elytron/src/main/java/org/wildfly/extension/elytron/DomainDefinition.java

```diff
@@ -550,5 +557,155 @@ public void execute(OperationContext context, ModelNode operation) throws Operat
+                    return;
+                }
+
+                SecurityIdentity identity = authenticationContext.getAuthorizedIdentity();
+                ModelNode result = context.getResult();
+
+                result.get(ElytronDescriptionConstants.NAME).set(principalName);
+
+                ModelNode attributesNode = result.get(ElytronDescriptionConstants.ATTRIBUTES);
+
+                identity.getAttributes().entries().forEach(entry -> {
+                    ModelNode entryNode = attributesNode.get(entry.getKey()).setEmptyList();
+                    entry.forEach(entryNode::add);
+                });
+
+                ModelNode rolesNode = result.get(ElytronDescriptionConstants.ROLES);
+                identity.getRoles().forEach(rolesNode::add);
+            } catch (RealmUnavailableException e) {
+                throw ROOT_LOGGER.couldNotReadIdentity(principalName, domainServiceName, e);
+            }
+        }
+    }
+
+    /**
+     * <p>A temporary operation that performs authentication based on a {@link SecurityDomain}. This operation will be removed once
+     * the subsystem is fully functional. It should be used for <em>test</em> purposes only.
+     *
+     * <p>This operation is very verbose in order to push messages back to CLI during tests.
+     */
+    static class AuthenticatorOperationHandler extends ElytronRuntimeOnlyHandler {
+
+        private static final ServiceUtil<SecurityDomain> DOMAIN_SERVICE_UTIL = ServiceUtil.newInstance(SECURITY_DOMAIN_RUNTIME_CAPABILITY, ElytronDescriptionConstants.SECURITY_DOMAIN, SecurityDomain.class);
+
+        private static final String OPERATION_NAME = "authenticate";
+        private static final String PARAMETER_USERNAME = "username";
+        private static final String PARAMETER_PASSWORD = "password";
+
+        public static final SimpleAttributeDefinition USER_NAME = new SimpleAttributeDefinitionBuilder(PARAMETER_USERNAME, ModelType.STRING, false)
+                .setAllowExpression(false)
+                .build();
+
+        public static final SimpleAttributeDefinition PASSWORD = new SimpleAttributeDefinitionBuilder(PARAMETER_PASSWORD, ModelType.STRING, false)
+                .setAllowExpression(false)
+                .build();
+
+        static String getOperationName() {
+            return OPERATION_NAME;
+        }
+
+        static AttributeDefinition[] getParameterDefinitions() {
+            return new AttributeDefinition[] {USER_NAME, PASSWORD};
+        }
+
+        public static void register(final ManagementResourceRegistration resourceRegistration, final ResourceDescriptionResolver resolver) {
+            resourceRegistration.registerOperationHandler(new SimpleOperationDefinition(AuthenticatorOperationHandler.getOperationName(), resolver,
+                    AuthenticatorOperationHandler.getParameterDefinitions()), new AuthenticatorOperationHandler());
```

Comment:
.setFlags(OperationEntry.Flag.HIDDEN)
.setRuntimeOnly()

The former will prevent this appearing in the described API, just in case it's not removed before .Final.